web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,184 @@
+        matchBackground: match,
+        activeMatchBackground: activeMatch,
+      },
+    });
+  }
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    search(e.target.value);
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      search(e.currentTarget.value);
+    }
+  };
+
+  function searchNext() {
+    search(searchValue);
+  }
+
+  function searchPrevious() {
+    search(searchValue, true /* search for previous result */);
+  }
+
+  const onEscape = useCallback(() => {
+    onClose();
+    xTerm.clearSearch();
+    xTerm.term.focus();
+  }, [xTerm, onClose]);
+
+  const onSearchResultsChange = useCallback(
+    (resultIndex: number, resultCount: number) => {
+      setSearchResults({ resultIndex, resultCount });
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!xTerm) {
+      return;
+    }
+    xTerm.loadSearchAddon({ onDidChangeResults: onSearchResultsChange });
+    const listener = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        onEscape();
+        e.preventDefault();
+      }
+      if ((e.metaKey || e.ctrlKey) && e.key === 'f') {
+        onOpen();
+        searchInputRef.current?.focus();
+        e.preventDefault();
+      }
+    };
+
+    document.addEventListener('keydown', listener);
+
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [onEscape, onOpen, searchInputRef, onSearchResultsChange, xTerm]);
+
+  if (!show) {
+    return;
+  }
+
+  return (
+    <SearchInputContainer alignItems="center" gap={3}>
+      <Input
+        ref={searchInputRef}
+        data-testid="terminal-search"
+        size="small"
+        autoFocus
+        onFocus={e => e.target.select()}
+        value={searchValue}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+      />
+      <P2>{`${searchResults.resultCount === 0 ? 0 : searchResults.resultIndex + 1}/${searchResults.resultCount}`}</P2>
+      <Divider />
+      <Flex gap={1}>
+        <SearchButtonIcon title="search next" onClick={searchNext}>
+          <ArrowDown size="small" />
+        </SearchButtonIcon>
+        <SearchButtonIcon title="search previous" onClick={searchPrevious}>
+          <ArrowUp size="small" />
+        </SearchButtonIcon>
+        <SearchButtonIcon title="close search" onClick={onEscape}>
+          <Cross size="small" />
+        </SearchButtonIcon>
+      </Flex>
+    </SearchInputContainer>
+  );
+};
+
+const SearchInputContainer = styled(Flex)`
+  max-width: 360px;
+  padding: ${p => p.theme.space[2]}px;
+  background-color: ${p => p.theme.colors.levels.surface};
+  border-radius: ${props => props.theme.radii[2]}px;
+`;
+
+const SearchButtonIcon = styled(ButtonIcon)`
+  border-radius: ${props => props.theme.radii[2]}px;
+`;
+
+const Divider = styled(Box)`
+  width: 2px;
+  height: 24px;
+  background-color: ${p => p.theme.colors.levels.popout};
+`;
+
+export interface TerminalWithSearch {
+  term: Xterm;
```

Comment:
We don't really need to install `@xterm/xterm` just for this type. We only use the `.focus()` method from it, we can simply add a prop `focusTerminal()` and get rid of the `term` prop.

Reply:
I also use it in the tests/story for mocks

Reply:
and now i dont! removed